I'm getting some gnarly error messages. I followed the instructions at https://github.com/pgcentralfoundation/pgrx#hacking.

- `export PATH="$HOME/pgrx/bin:$PATH"` is in `~/.bashrc`.
- I have this repository on disk at `~/pgrx`.

```
TRYBUILD=overwrite cargo test --features pg14
```

starts out fine, but quickly deteriorates into

<details>
<summary>Error messages</summary>

```
error[E0603]: function `main` is private
   --> /home/hamir/pgrx/pgrx/src/lib.rs:319:29
    |
319 |         pub use pgrx_embed::main;
    |                             ^^^^ private function
    |
note: the function `main` is defined here
   --> /tmp/.tmp9wRy3Q:1:1
    |
1   | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let contro...
    | ^^^^^^^^^^

error[E0601]: `main` function not found in crate `pgrx_embed_pgrx_tests`
 --> pgrx-tests/./src/bin/pgrx_embed.rs:1:23
  |
1 | ::pgrx::pgrx_embed!();
  | --------------------- ^ the main function must be defined at the crate level (in `pgrx-tests/./src/bin/pgrx_embed.rs`)
  | |
  | non-function item at `crate::main` is found
  |
note: here is a function named `main`
 --> /tmp/.tmp9wRy3Q:1:1
  |
1 | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let control_...
  | ^^^^^^^^^^
  = note: you have one or more functions named `main` not defined at the crate level
  = help: consider moving the `main` function definitions

Some errors have detailed explanations: E0601, E0603.
For more information about an error, try `rustc --explain E0601`.
error: could not compile `pgrx-tests` (bin "pgrx_embed_pgrx-tests") due to 2 previous errors
test tests::aggregate_tests::tests::pg_aggregate_demo_custom_state ... FAILED
       Using PgConfig("pg14") and `pg_config` from /home/hamir/.pgrx/14.13/pgrx-install/bin/pg_config
    Building extension with features pg14 pg_test proptest
     Running command "/home/hamir/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo" "build" "--lib" "--features" "pg14 pg_test proptest" "--no-default-features" "--message-format=json-render-diagnostics"
warning: creating a shared reference to mutable static is discouraged
   --> pgrx-tests/src/tests/memcxt_tests.rs:112:28
    |
112 |                 assert_eq!(pg_sys::CurrentMemoryContext, ctx_sys);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
    = note: `#[warn(static_mut_refs)]` on by default

warning: `pgrx-tests` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
  Installing extension
     Copying control file to /home/hamir/.pgrx/14.13/pgrx-install/share/postgresql/extension/pgrx_tests.control
     Copying shared library to /home/hamir/.pgrx/14.13/pgrx-install/lib/postgresql/pgrx_tests.so
  Discovered 948 SQL entities: 58 schemas (6 unique), 832 functions, 22 types, 2 enums, 6 sqls, 1 ords, 1 hashes, 9 aggregates, 17 triggers
  Rebuilding for SQL generation with features `pg14 pg_test proptest`
warning: creating a shared reference to mutable static is discouraged
   --> pgrx-tests/src/tests/memcxt_tests.rs:112:28
    |
112 |                 assert_eq!(pg_sys::CurrentMemoryContext, ctx_sys);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
    = note: `#[warn(static_mut_refs)]` on by default

warning: `pgrx-tests` (lib) generated 1 warning
   Compiling pgrx-tests v0.12.7 (/home/hamir/pgrx/pgrx-tests)
error[E0603]: function `main` is private
   --> /home/hamir/pgrx/pgrx/src/lib.rs:319:29
    |
319 |         pub use pgrx_embed::main;
    |                             ^^^^ private function
    |
note: the function `main` is defined here
   --> /tmp/.tmpkcVQsp:1:1
    |
1   | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let contro...
    | ^^^^^^^^^^

error[E0601]: `main` function not found in crate `pgrx_embed_pgrx_tests`
 --> pgrx-tests/./src/bin/pgrx_embed.rs:1:23
  |
1 | ::pgrx::pgrx_embed!();
  | --------------------- ^ the main function must be defined at the crate level (in `pgrx-tests/./src/bin/pgrx_embed.rs`)
  | |
  | non-function item at `crate::main` is found
  |
note: here is a function named `main`
 --> /tmp/.tmpkcVQsp:1:1
  |
1 | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let control_...
  | ^^^^^^^^^^
  = note: you have one or more functions named `main` not defined at the crate level
  = help: consider moving the `main` function definitions

Some errors have detailed explanations: E0601, E0603.
For more information about an error, try `rustc --explain E0601`.
error: could not compile `pgrx-tests` (bin "pgrx_embed_pgrx-tests") due to 2 previous errors
```
</details>

repeated a lot of times.

--- a/pgrx-tests/tests/compile-fail/table-iterators-arent-immortal.stderr
+++ b/pgrx-tests/tests/compile-fail/table-iterators-arent-immortal.stderr
@@ -1,3 +1,15 @@
+warning: elided lifetime has a name
+ --> tests/compile-fail/table-iterators-arent-immortal.rs:6:19
+  |
+6 | ) -> TableIterator<(name!(a, &'static str), name!(b, Option<&'static str>))> {
+  |                   ^ this elided lifetime gets resolved as `'static`
+  |
+  = note: `#[warn(elided_named_lifetimes)]` on by default
+help: consider specifying it explicitly
+  |
+6 | ) -> TableIterator<'static, (name!(a, &'static str), name!(b, Option<&'static str>))> {
+  |                    ++++++++
+
 error[E0521]: borrowed data escapes outside of function
  --> tests/compile-fail/table-iterators-arent-immortal.rs:6:78
   |

